packages/quantic/force-app/main/default/lwc/quanticCaseClassification/e2e/quanticCaseClassification.e2e.ts

```diff
@@ -0,0 +1,269 @@
+        const caseClassificationsResponsePromise =
+          caseAssist.waitForCaseClassificationsResponse();
+        await caseAssist.fetchClassifications();
+        await caseClassificationsResponsePromise;
+        await expect(caseClassification.selectedSuggestion).toHaveValue(
+          caseClassificationSuggestions[selectedSuggestionIndex].value
+        );
+      });
+    });
+
+    test.describe('when selecting a specific suggestion and then receiving new suggestions that does not contain the previously selected option', () => {
+      test('should keep the suggestion selected by the user and display it in the select input', async ({
+        caseClassification,
+        caseAssist,
+      }) => {
+        const selectedSuggestionIndex = 1;
+        await caseClassification.clickCaseClassificationSuggestionAtIndex(
+          selectedSuggestionIndex
+        );
+        await expect(caseClassification.selectedSuggestion).toHaveValue(
+          caseClassificationSuggestions[selectedSuggestionIndex].value
+        );
+
+        caseClassification.mockCaseClassification(coveoDefaultField, [
+          caseClassificationSuggestions[0],
+        ]);
+        const caseClassificationsResponsePromise =
+          caseAssist.waitForCaseClassificationsResponse();
+        await caseAssist.fetchClassifications();
+        await caseClassificationsResponsePromise;
+
+        await expect(caseClassification.allOptionsSelectInput).toHaveText(
+          caseClassificationSuggestions[selectedSuggestionIndex].value
+        );
+      });
+    });
+
+    test.describe('when selecting an option from the select input and then fetching suggestions', () => {
+      test('should keep the option selected by the user, display it in the select input and hide the suggestions', async ({
+        caseClassification,
+        caseAssist,
+      }) => {
+        const selectedOptionIndex = 3;
+        await caseClassification.clickShowSelectInputButton();
+        await caseClassification.clickallOptionsSelectInput();
+        await caseClassification.clickSelectInputOptionAtIndex(
+          selectedOptionIndex
+        );
+
+        caseClassification.mockCaseClassification(coveoDefaultField, [
+          caseClassificationSuggestions[0],
+        ]);
+        const caseClassificationsResponsePromise =
+          caseAssist.waitForCaseClassificationsResponse();
+        await caseAssist.fetchClassifications();
+        await caseClassificationsResponsePromise;
+        await expect(caseClassification.allOptionsSelectInput).toHaveText(
+          allOptions[selectedOptionIndex].value
+        );
+        await expect(
+          caseClassification.caseClassificationSuggestions
+        ).toHaveCount(0);
+      });
+    });
+
+    test.describe('when receiving new suggestions without changing the by default auto-selected suggestion', () => {
+      test('should auto-select the suggestion with the highest confidence from the newly received suggestions', async ({
```

Comment:
This also could perhaps be a unit test?

Reply:
same: https://github.com/coveo/ui-kit/pull/4967#discussion_r1973683527